chore: update the Miden VM (with updated `miden-package` crate)

greenhat opened the pull request:
This PR updates the Miden VM to https://github.com/0xPolygonMiden/miden-vm/tree/greenhat/miden-package-pre-eam-vm-for-compiler (the tip of the https://github.com/0xPolygonMiden/miden-vm/pull/1544 before rebasing on EAM version) and can be merged now.

bobbinth commented:
I'm planning to merge https://github.com/0xPolygonMiden/miden-vm/pull/1631 relatively soon (which will probably have impact on this PR).

greenhat commented:
> I'm planning to merge [0xPolygonMiden/miden-vm#1631](https://github.com/0xPolygonMiden/miden-vm/pull/1631) relatively soon (which will probably have impact on this PR).

Thanks for the heads-up! As long as we can pin to a commit with my PR changes before the #1631 merge, we should be fine.

I'd prefer to migrate to the memory-addressable VM version after we got the frontend migrated and the new IR pipeline ironed out. I made #374 for this.

greenhat commented:
@bitwalker I rebased this branch and updated the VM to the version of my branch before it was rebased onto the EAM VM version (see link in the PR descrioption).
